tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,505 @@
+	configMapData := DefineBaseConfig(tsparams.DaemonsFile, "", "")
+	hubConfigMap, err := configmap.NewBuilder(APIClient, name, tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create hub config map")
+
+	return hubConfigMap
+}
+
+func createMasterConfigMap(name, configMapStaticRoutes string) *configmap.Builder {
+	configMapData := DefineBaseConfig(tsparams.DaemonsFile, configMapStaticRoutes, "")
+	hubConfigMap, err := configmap.NewBuilder(APIClient, name, tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create hub config map")
+
+	return hubConfigMap
+}
+
+func createFrrHubPod(name, nodeName, configmapName string, defaultCMD []string,
+	secondaryNetConfig []*types.NetworkSelectionElement) *pod.Builder {
+	frrPod := pod.NewBuilder(APIClient, name, tsparams.TestNamespaceName, NetConfig.FrrImage).
+		DefineOnNode(nodeName).
+		WithTolerationToMaster().
+		WithSecondaryNetwork(secondaryNetConfig).
+		RedefineDefaultCMD(defaultCMD)
+
+	By("Creating FRR hub pod in the test namespace")
+
+	frrContainer := pod.NewContainerBuilder(
+		"frr2", NetConfig.CnfNetTestContainer, []string{"/bin/bash", "-c", "sleep INF"}).
+		WithSecurityCapabilities([]string{"NET_ADMIN", "NET_RAW", "SYS_ADMIN"}, true)
+
+	frrCtr, err := frrContainer.GetContainerCfg()
+	Expect(err).ToNot(HaveOccurred(), "Failed to get container configuration")
+	frrPod.WithAdditionalContainer(frrCtr).WithLocalVolume(configmapName, "/etc/frr")
+
+	frrPod, err = frrPod.WithPrivilegedFlag().CreateAndWaitUntilRunning(5 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create FRR test pod")
+
+	return frrPod
+}
+
+func createMasterPodTest(name, nodeName, configmapName string, defaultCMD []string,
+	secondaryNetConfig []*types.NetworkSelectionElement, portNum int) *pod.Builder {
+	frrPod := pod.NewBuilder(APIClient, name, tsparams.TestNamespaceName, NetConfig.FrrImage).
+		DefineOnNode(nodeName).
+		WithTolerationToMaster().
+		WithSecondaryNetwork(secondaryNetConfig).RedefineDefaultCMD(defaultCMD)
+
+	By("Creating FRR master container in the test namespace")
+
+	frrContainer := pod.NewContainerBuilder(
+		containerFrr, NetConfig.CnfNetTestContainer, []string{"/bin/bash", "-c",
+			fmt.Sprintf("testcmd -interface net1 -protocol tcp -port %d -listen", portNum)}).
+		WithSecurityCapabilities([]string{"NET_ADMIN", "NET_RAW", "SYS_ADMIN"}, true)
+
+	frrCtr, err := frrContainer.GetContainerCfg()
+	Expect(err).ToNot(HaveOccurred(), "Failed to get container configuration")
+	frrPod.WithAdditionalContainer(frrCtr).WithLocalVolume(configmapName, "/etc/frr")
+
+	frrPod, err = frrPod.WithPrivilegedFlag().CreateAndWaitUntilRunning(5 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create FRR test pod")
+
+	return frrPod
+}
+
+func setWorkerNodeListAndLabelForTests(workerNodeList []*nodes.Builder) (map[string]string, []*nodes.Builder) {
```

Comment:
What does this func?

Reply:
Its returnig workerLabelMap, workerNodeList

Reply:
So why do we need it??  it gets workerNodeList and returns it. and NetConfig.WorkerLabelMap you can take directly if needed